pages/pil/dashboard/index.js

```diff
@@ -8,24 +7,23 @@ module.exports = settings => {
     root: __dirname
   });
 
-  const profileOwnsModules = (profile, modules) => {
-    const profileModuleIds = profile.trainingModules.map(module => module.id);
-    return modules.every(id => profileModuleIds.includes(id));
+  const profileOwnsModule = (profile, moduleId) => {
+    return !!profile.trainingModules.find(module => module.id === moduleId);
   };
 
   app.post('/', bodyParser.urlencoded({ extended: true }), (req, res, next) => {
-    if (req.body.action === 'delete' && req.body.modules.length) {
+    if (req.body.action === 'delete' && req.body.trainingModuleId) {
       const profile = res.locals.model;
-      const modules = req.body.modules;
+      const trainingModuleId = req.body.trainingModuleId;
 
-      if (!profileOwnsModules(profile, modules)) {
+      if (!profileOwnsModule(profile, trainingModuleId)) {
         throw new Error('cannot delete training modules the profile does not own');
       }
 
       const opts = {
         method: 'DELETE',
         headers: { 'Content-type': 'application/json' },
-        body: JSON.stringify({ modules })
+        body: JSON.stringify({ id: trainingModuleId })
```

Comment:
FYI - you can shortcut the headers and the `JSon.stringify` by passing a `json` property.

e.g. 
```js
opts = {
  method: 'DELETE',
  json: { id: trainingModuleId }
};
```